src/compute/src/compute_state.rs

```diff
@@ -294,6 +299,21 @@ impl ComputeState {
         // Remember the maintenance interval locally to avoid reading it from the config set on
         // every server iteration.
         self.server_maintenance_interval = COMPUTE_SERVER_MAINTENANCE_INTERVAL.get(config);
+
+        if self.replica_expiration.is_none() {
+            let offset = COMPUTE_REPLICA_EXPIRATION.get(&self.worker_config);
+            if !offset.is_zero() {
+                // TODO(sdht0): consistently send now() from environmentd.
+                let now = mz_ore::now::SYSTEM_TIME.clone()();
```

Comment:
Sending now() consistently from environmentd should probably be part of this PR before we test in production.

Reply:
I don't think so. We can add more slack to the configured timeout, as long as it's larger than the biggest time delta we're expecting it'll be fine. Broadcasting the current time from envd is itself a difficult problem, and I don't want to make it part of this PR.

Reply:
Noted.

Reply:
I'd even say there is no need to broadcast `now` from envd even as a follow up, at least not for this feature. There is no reason for why consistent times would be beneficial here, AFAICT, so the additional complexity is not worth it. So I suggest dropping that `TODO`.

Reply:
Alright, updated!